Pontoon: browsing and search (fixes issue #36)

flodolo opened the pull request:
Fixes issue #36

flodolo commented:
@mathjazz 
I'm stuck on the definition of Untranslated…

flodolo commented:
I've split the original doc in two parts. This somehow depends on #38, since these pages need to be added to the index, and the user permission link is broken until that one is merged.

flodolo commented:
> @mathjazz
> I'm stuck on the definition of Untranslated…

OK, clarified on IRC. I think this one is ready for review. As explained, it should be merged after #38, and one link is broken until that happens.

flodolo commented:
Probably easier to review from here
https://github.com/mozilla-l10n/localizer-documentation/tree/issue36/tools/pontoon

Two new files are currently orphaned, will need to add them to index and summary for GitBook.

UncleNachoDuh commented on tools/pontoon/search_filters.md:
```diff
@@ -0,0 +1,30 @@
+* *Missing*: string is not available in the localized file.
+* *Suggested*: at least one translation has been suggested for this string. Note that, in this case, the translation only exists within Pontoon database, it’s not exported to external repositories.
+* *Translated*: string has an approved translation.
+* *Fuzzy*: string is marked as fuzzy in the localized file. This state is possible only if the source is a [Gettext](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html) (.po) file. Translation is in the file, marked as fuzzy and therefore not used by the website or application.
```
For the new contributor, it may not be clear what "fuzzy" means here. It is clear that the translation exists in Pontoon and not in the website or app, but not what "fuzzy" means. 

Are we concerned with making this explicit in this doc?

flodolo replied:
To be honest I was hoping to get away with the doc to Gettext, maybe we can add a link to https://www.gnu.org/software/gettext/manual/html_node/Fuzzy-Entries.html#Fuzzy-Entries

UncleNachoDuh replied:
I think that would be great. It would be a useful link to new localizers. When I first started contributing on Pootle, I didn't know the term fuzzy match. Jeff taught it to me :).

flodolo replied:
Added https://github.com/mozilla-l10n/localizer-documentation/pull/39/commits/3eb95ef5a06c6c2412c5bc9c839d240bf4634635

flodolo commented:
> From my experience with documentation, I feel like I would want to link out to something at the end of this doc. I am not sure about how this documentation is meant to flow, but I am imagining the flow to be like an on-boarding experience, and therefore I am looking for what's next. Maybe that's a link to the teams doc or the editing in Pontoon doc.
> 
> What do you think?

I think this should be more like a manual, with stand-alone documents. There's an index linking them, internal references, plus search (on https://mozilla-l10n.github.io/localizer-documentation/).

It becomes quite hard to link them onboarding style, especially considering how diverse topics are across the entire project.

But let's also see what other folks think.

flodolo commented:
> I have one suggestion for the search documentation about the definition of "fuzzy"

For better visibility, since the comment is in an outdated piece of code: addressed in https://github.com/mozilla-l10n/localizer-documentation/commit/3eb95ef5a06c6c2412c5bc9c839d240bf4634635

mathjazz commented on tools/pontoon/team_project.md:
```diff
@@ -0,0 +1,68 @@
+
+![Team page for Italian](/assets/images/pontoon/team_project/team_page.png)
+
+The page header contains linguistic information about this locale (plural form, writing direction, etc.), plus an indication of the overall status of completion and statistics.
```
Nit: this locale -> the locale

mathjazz commented on tools/pontoon/team_project.md:
```diff
@@ -0,0 +1,68 @@
+* **Info**: information about the team. This is currently only editable by administrators ([bug 1343804](https://bugzilla.mozilla.org/show_bug.cgi?id=1343804)).
+* **Permissions**: manage user permissions (see [this document](users.md#managing-permissions) for details).
+
+Hovering a project in the list will replace the progress bar with a set of detailed statistics (untranslated strings, missing strings, etc.). Note that all these numbers are links, so you can use them to open the project with a filter already enabled, for example to display only missing strings. You can click on the number of total strings to access *All Resources* (i.e. all files) for this proje ...
```
all files -> all strings in all files

mathjazz commented on tools/pontoon/team_project.md:
```diff
@@ -0,0 +1,68 @@
+* **Project info**: information about the project.
+* **Team info**: same content as the *Info* tab in the the team page.
+
+As for [Teams page](https://pontoon.mozilla.org/teams/), there’s also a landing [Projects page](https://pontoon.mozilla.org/projects/). From this page it’s possible to access each project page, to see the overall translation status, or when was the last time that each locale worked on the project.
```
As for -> Similar to the

mathjazz commented on tools/pontoon/team_project.md:
```diff
@@ -0,0 +1,68 @@
+
+![Project hover](/assets/images/pontoon/team_project/project_hover.png)
+
+## Project page
```
This paragraph is actually about what's called a "Localization page" in Pontoon. It captures data of the particular project localized into particular locale.

The "Project page" would be what's on the Mozilla.org screenshot at the end of the paragraph.

mathjazz commented on tools/pontoon/team_project.md:
```diff
@@ -0,0 +1,68 @@
+
+There are 4 tabs available (highlighted in red):
+* **Resources**: files available in the project.
+* **Contributors**: a list of active contributors with their statistics, filterable by time. Unlike the tab available in the team page, this only includes contributions to this specific project.
```
contributions to this **localization** (of a particular project to a particular locale).

flodolo commented on tools/pontoon/search_filters.md:
```diff
@@ -0,0 +1,30 @@
+# Search and filters
+
+A string in Pontoon can be in one of these states:
+* *Missing*: string is not available in the localized file.
```
Uhm, that's not true, a fuzzy string **is** in the file, there's no way to mark a string as fuzzy in Pontoon, right?

Using for now: `string is not available in the localized file and doesn’t have any translations or suggestions in Pontoon`

flodolo commented:
Thanks @mathjazz, turns out I had a very confused idea about Pontoon's pages.

Fixes for the search page are somehow easy, and they're here
https://github.com/mozilla-l10n/localizer-documentation/pull/39/commits/7d464fc6f7ccaba53d0e3976ebe10adae36a6ffe

Fixes on the other pages are definitely more challenging.
* I've renamed the file
* I don't think Teams and Projects pages deserve a section, I've added references in an intro section.

Hopefully discussion from here will be easier
https://github.com/mozilla-l10n/localizer-documentation/pull/39/commits/591e4e50a64f1e1b8ae11e2ac544299b971296a7

flodolo commented:
For the other question (thumbnails). I think you'd need to manually generate them, and link them.
On the other hand, I'm not a huge fan of having to click a thumbnail to actually see it.

I guess we'll need to generate the pages on GitBook and see how it looks, and eventually address all images/thumbnails in a different issue.

mathjazz commented on tools/pontoon/teams_projects.md:
```diff
@@ -0,0 +1,76 @@
+
+## Project page
+
+Similar to the [Teams page](https://pontoon.mozilla.org/teams/), there’s also a landing [Projects page](https://pontoon.mozilla.org/projects/). From this page it’s possible to access each *Project page*, to see the overall translation status, or when was the last time that each locale worked on the project.
```
The Projects page has already been introduced on top of the document.

Instead of this paragraph, we should introduce the Project page, similarly as we introduce Team and Localization pages:
1. How to access it.
2. Screenshot.
3. Header.
4. Tabs

flodolo commented:
Thanks, added https://github.com/mozilla-l10n/localizer-documentation/pull/39/commits/e690d1c47605956b06dc39ff9db9688d2f0f8078.

I'll work on the team/project page next week at this point.

flodolo commented:
@mathjazz 
Updated the other page too in https://github.com/mozilla-l10n/localizer-documentation/pull/39/commits/726804fd03d2d4a4b69336bd54d9100eff1b6dae
https://github.com/mozilla-l10n/localizer-documentation/blob/issue36/tools/pontoon/teams_projects.md

I renamed also the images folder for consistency with the page name.

mathjazz commented on tools/pontoon/teams_projects.md:
```diff
@@ -0,0 +1,88 @@
+* **Deadline**: set only for some projects, it indicates when the translation is due.
+* **Repository**: link to the repository storing translations for this project.
+* **Resources**: links to useful external resources. Custom links can be defined for each project.
+* **Contact person**: localization driver in charge of this project.
```
I'd move this seciton under the Project page, because that's where users will see it first (in this document and during Pontoon use).

flodolo replied:
Thanks, swapped them.
